packages/notification/src/features/email/service..ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+
+import NotificationQueue, {
+  NotificationQueueRecord
+} from '@notification/model/notificationQueue'
+import { logger } from '@notification/logger'
+import { notifyCountryConfig } from '@notification/features/sms/service'
+import { internal } from '@hapi/boom'
+import * as Hapi from '@hapi/hapi'
+
+interface AllUsersEmailPayloadSchema {
+  subject: string
+  body: string
+  bcc: string[]
+  locale: string
+}
+
+let isLoopInprogress = false
+
+export async function sendAllUserEmails(request: Hapi.Request) {
+  const payload = request.payload as AllUsersEmailPayloadSchema
+  await NotificationQueue.create(payload)
+  if (!isLoopInprogress) {
+    loopNotificationQueue(request)
+  }
+  return {
+    success: true
+  }
```

Comment:
You can probably implement the "you've already sent an email today" (if this is important) by modifying `deleteRecord` to only delete records that are older than one day (or were created the previous date). 
To know when the user has already sent something today, you'd just look at the database – if it has records created today (either successfully sent or failed), then you let the user know they've already sent something

Reply:
According to the current approach, we delete the record after the mails have been successfully sent. This approach needs the succeeded message queue item to be stored. And at some point those need to be deleted

Reply:
Yea, I guess it's ok to always keep one day's worth of records in the database if it allow us to implement this functionality without having to use another database for instance